roc/sender.go

```diff
@@ -1,9 +1,95 @@
 package roc
 
 /*
-#include <roc/sender.h>
+ #include <roc/sender.h>
+int roc_sender_write_frame(roc_sender* sender, float* samples, unsigned long samples_size) {
+    roc_frame frame = {(void*)samples, samples_size};
+    return roc_sender_write(sender, &frame);
+ }
 */
 import "C"
 
+import (
+	"fmt"
+	"unsafe"
+)
+
 // Sender as declared in roc/sender.h:96
 type Sender C.roc_sender
+
+func OpenSender(rocContext *Context, senderConfig *SenderConfig) (*Sender, error) {
+	senderConfigC := C.struct_roc_sender_config{
+		frame_sample_rate:        (C.uint)(senderConfig.FrameSampleRate),
+		frame_channels:           (C.roc_channel_set)(senderConfig.FrameChannels),
+		frame_encoding:           (C.roc_frame_encoding)(senderConfig.FrameEncoding),
+		packet_sample_rate:       (C.uint)(senderConfig.PacketSampleRate),
+		packet_channels:          (C.roc_channel_set)(senderConfig.PacketChannels),
+		packet_encoding:          (C.roc_packet_encoding)(senderConfig.PacketEncoding),
+		packet_length:            (C.ulonglong)(senderConfig.PacketLength),
+		packet_interleaving:      (C.uint)(senderConfig.PacketInterleaving),
+		automatic_timing:         (C.uint)(senderConfig.AutomaticTiming),
+		resampler_profile:        (C.roc_resampler_profile)(senderConfig.ResamplerProfile),
+		fec_code:                 (C.roc_fec_code)(senderConfig.FecCode),
+		fec_block_source_packets: (C.uint)(senderConfig.FecBlockSourcePackets),
+		fec_block_repair_packets: (C.uint)(senderConfig.FecBlockRepairPackets),
+	}
+	sender := C.roc_sender_open((*C.roc_context)(rocContext), &senderConfigC)
+	if sender == nil {
+		return nil, ErrInvalidArgs
+	}
+	return (*Sender)(sender), nil
+}
+
+func (s *Sender) Bind(a *Address) error {
+	errCode := C.roc_sender_bind((*C.roc_sender)(s), a.raw)
+	if errCode == 0 {
+		return nil
+	}
+	if errCode < 0 {
+		return ErrInvalidArgs // roc_receiver_bind returns negative in 3 cases: invalid args, cannot bind address and not enough resources, maybe we should consider renaming this error to a more generic name?
+	}
+	panic(fmt.Sprintf(
+		"unexpected return code %d from roc_address_init()", errCode))
+}
+
+func (s *Sender) Connect(portType PortType, proto Protocol, address *Address) error {
+	errCode := C.roc_sender_connect(
+		(*C.roc_sender)(s),
+		(C.roc_port_type)(portType),
+		(C.roc_protocol)(proto),
+		address.raw,
+	)
+	if errCode == 0 {
+		return nil
+	}
+	if errCode < 0 {
+		return ErrInvalidArgs // roc_receiver_bind returns negative in 3 cases: invalid args, cannot bind address and not enough resources, maybe we should consider renaming this error to a more generic name?
+	}
+	panic(fmt.Sprintf(
+		"unexpected return code %d from roc_address_init()", errCode))
+}
+
+func (s *Sender) WriteFloats(frame []float32) error {
+	sizeOfFloat := (int)(unsafe.Sizeof(frame[0]))
+	errCode := C.roc_sender_write_frame((*C.roc_sender)(s), (*C.float)(unsafe.Pointer(&frame[0])), (C.ulong)(len(frame)*sizeOfFloat))
```

Comment:
Consider moving sizeof into the jumper function (since it's already knows the sample type (float)).

Also, the unsafe.Pointer here is unnecessary.